File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/Abu_HadHoud_Solutions_Using_C++/Problem3.cpp
#include <iostream>
using namespace std;

enum enNumberType { odd = 1, even =2 };
int ReadNumber() {
	int num;
	cout << "Please enter your  number : ";
	cin >> num;
	return num;
}
enNumberType CheckNumberType(int num) {
	int Result = num % 2;
	if (Result == 0) {
		return enNumberType::even;
	}
	else {
		return enNumberType::odd;
	}

}
void PrintNumberType(enNumberType NumberType) {
	if (NumberType == enNumberType::even) 
		cout << "\n Number is Even. \n";
	else 
		cout << "\n Number is Odd. \n";
	
}
int main() {
	PrintNumberType(CheckNumberType(ReadNumber()));
}

File: Course 04 [Algorithms & Problem-Solving Level 1 Solutions]/My_Solutions_Using_C#/Problem3.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Threading;
using System.Threading.Tasks;

namespace AbuHadHoud
{
    internal class Program
    {
        enum enNumberType { odd = 1, even = 2 }
        static int ReadNum()
        {
            int num;
            Console.Write("Please enter your Number : ");
            num = Int32.Parse(Console.ReadLine());
            return num;
        }
        static enNumberType CheckNumberType(int num)
        {
            if (num % 2 == 0)
                return enNumberType.even;
            else
                return enNumberType.odd;
        }
        static void PrintNumberType(enNumberType NumberType)
        {
            if(NumberType == enNumberType.even)
                Console.WriteLine("\nNumber is Even.");
            else
                Console.WriteLine("\nNumber is Odd.");
        }

        static void Main(string[] args)
        {
            PrintNumberType((CheckNumberType(ReadNum())));
        }
    }
}